apps/emqx/src/emqx_router.erl

```diff
@@ -251,7 +251,7 @@ delete_trie_route(Route = #route{topic = Topic}) ->
 %% @private
 -spec(maybe_trans(function(), list(any())) -> ok | {error, term()}).
 maybe_trans(Fun, Args) ->
-    case persistent_term:get(emqx_route_lock_type) of
+    case emqx_config:get([broker, perf, route_lock_type]) of
```

Comment:
pls hold. let me check the consequence of this change.

Reply:
I'd prefer keep this implementation now. Later if there's any issue found by benchmarking I will change it then.